### DISCUSSION TITLE
Adding more machine translation alternatives

### DISCUSSION THREAD

**Discussion by Prasanta-Hembram** [immersive-translate/immersive-translate#1358] [Ideas]
### 新功能 | New Feature

1. https://translate.glosbe.com (Glosbe Translate hosted by Glosbe)
2. https://translate.wmcloud.org/ (MinT Machine translation hosted by Wikimedia Foundation) - API spec (https://translate.wmcloud.org/api/spec)
3. Mymemory api (https://mymemory.translated.net/doc/spec.php)

### 为什么您想要添加这些功能 | Why you want to add these features

1. Adding these machine translation alternatives will increase the number of languages supported for machine translation up to [200](https://ai.facebook.com/research/no-language-left-behind/), including low-resource languages such as Asturian and Luganda. These online hosted machine translation service providers use NLLB (both Glosbe and MinT) and MarianNMT (only Glosbe) models.

### 提议 | Proposal

Adding support to more machine servies to extention.

**Comment by theowenyoung** [comment]
Hello, at the moment you can use the Custom API to incorporate any translation service: https://immersivetranslate.com/docs/services/custom/

We plan to provide an optional translation service in the future, where more translation engines can be supported
